src/utils/log-file-system.ts

```diff
@@ -0,0 +1,29 @@
+import { REPO_DIR } from "../global-state"
+import LightningFS from "@isomorphic-git/lightning-fs"
+
+export async function logFileSystemState(fs: LightningFS, dir = REPO_DIR) {
```

Comment:
This looks handy! I think it makes sense to move this into the `utils/fs.ts` file.